src/groupby.jl

```diff
@@ -587,3 +590,16 @@ function dataset(gr::ReducedGroupedVariable)
         gr.reduce_fun,
     )
 end
+
+# ReducedGroupedVariable is a Variable and therefore an AbstractDiskArray.
+# getindex is overloaded for ReducedGroupedVariable and therefore 
+# ReducedGroupedVariable is defined to use getindex.
+# An alternative solution is to remove getindex definitions 
+# and then implement the read logic in readblock!
+function DiskArrays.readblock!(gr::ReducedGroupedVariable{T,N},
+    aout,
+    indexes::Vararg{OrdinalRange, N}) where {T,N}
+
+    aout .= Base.getindex(gr,indexes...)
```

Comment:
Currently `Base.getindex` is defined for `ReducedGroupedVariable` so the DiskArray.jl implementation is not used. An alternative would be to use the DiskArray `getindex` and implement more logic in `readblock!`.  This is more `ReducedGroupedVariable` and I can not see any benefit of doing it right now.